### DISCUSSION TITLE
How do I check that a EVP_PKEY has no value assigned?

### DISCUSSION THREAD

**Discussion by beldmit** [openssl/openssl#22271] [Q&A] [answered]
I'd like to distinguish a freshly created (via `EVP_PKEY_new`) and initialized (both via keymgmt and `EVP_PKEY_set_*` functions) calls. How can I do it?

**Comment by paulidale** [comment]
Does the `EVP_PKEY_is_a()` rule things out suitable?

**Comment by levitte** [comment, marked as answer]
The answer is `EVP_PKEY_get_id()`.  If it returns `EVP_PKEY_NONE`, then the pkey is unassigned.

We really should have a convenience function like `EVP_PKEY_is_assigned()`
